### PR TITLE
Export typings with the NPM bundle

### LINKED ISSUE #602: [FEATURE] Exports typings together with npm bundle
**Is your feature request related to a problem? Please describe.**
Using `design-reat-kit` in Typescript could be problematic.
Currently you have to compile them, then copy the file manually in the module folder. Not a great experience.

**Describe the solution you'd like**
Having type definition file installed together with the bundle when in a Typescript project

**Describe alternatives you've considered**
* Ship the typings as separate bundle.
  * Possible but we have them already here, and it would be better to mantain them together with the codebase.

### PR DESCRIPTION
**Pull request by sanyamdogra**
Fixes #602 

#### PR Checklist

- [x] My branch is up-to-date with the Upstream `master` branch.
- [x] The unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

#### Short description of what this resolves:
Export typings with the NPM bundle

#### Changes proposed in this pull request:

- Added steps for bundling typings with npm_publish

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -5,14 +5,13 @@
...
+  - &working_directory ~/repo
 
   - &restore_cache
     keys:
-    - npm-dependencies-v4-{{ checksum "package.json" }}
-    # fallback to using the latest cache if no exact match is found
-    - npm-dependencies-v4-
+      - npm-dependencies-v4-{{ checksum "package.json" }}
+      # fallback to using the latest cache if no exact match is found
+      - npm-dependencies-v4-
```
Comment (dej611) [inline .circleci/config.yml:14]: Please restore this as it was before

File: .circleci/config.yml [hunk 2]
```diff
@@ -115,10 +139,15 @@ workflows:
           filters:
             branches:
               ignore:
-                  - gh-pages
-                  - next
+                - gh-pages
+                - next
```
Comment (dej611) [inline .circleci/config.yml:143]: Please restoree as it was

**Review by dej611 — CHANGES_REQUESTED** [review]
The build is currently failing on CircleCI. The formatting of the file seems not alright.
Can you restore the previous parts as they were and just add the new workflow?

File: .circleci/config.yml (hunk 1 above)
Comment (sanyamdogra) [reply .circleci/config.yml]: Done

File: .circleci/config.yml (hunk 2 above)
Comment (sanyamdogra) [reply .circleci/config.yml]: Done

**Review by dej611 — APPROVED** [review]

**Pull request merged**